### PR TITLE
MDEV-30203: create compat packages

### PR DESCRIPTION
**Pull request by grooverdan**
eNote: strictly requires #2382 as without it, packages without compat won't function correctly.

Addresses last review: https://lists.launchpad.net/maria-developers/msg13270.html

- [x] *The Jira issue number for this PR is: MDEV-30203*

## Description

For both Deb and RPM, create mariadb-client-compat and mariadb-server-compat containing the mysql links to the mariadb named executables/scripts.

The mariadb-client-core mysqlcheck was moved to mariadb-client-compat.

resolve-stack-dump was moved from server to client (RPM), like where the man page and Debian package put it.

The symlinks in MYSQL_ADD_EXECUTABLE is tagged as a {Client,Server}Symlinks component and placed in
the symlinks packages.

Man pages are restructured be installed into compat package if that matches the executable.

Plugins now can have their own manual pages, as has been done for rocksdb and s3.

Columnstore has a workaround as it doesn't use the cmake/plugin.cmake.

Additionally, without WSREP, no galera man pages are installed.

Scripts likewise have compatibility symlinks are in the {server,client}-compat packages.

## How can this PR be tested?

[Salsa version pipeline combined with #2382](https://salsa.debian.org/grooverdan/mariadb-server/-/pipelines/473457).

### CODE REVIEW

File: extra/CMakeLists.txt [hunk 1]
```diff
@@ -91,7 +91,7 @@ MYSQL_ADD_EXECUTABLE(replace replace.c COMPONENT Client)
 TARGET_LINK_LIBRARIES(replace mysys)
 
 IF(UNIX)
-  MYSQL_ADD_EXECUTABLE(resolve_stack_dump resolve_stack_dump.c)
+  MYSQL_ADD_EXECUTABLE(resolve_stack_dump resolve_stack_dump.c COMPONENT Client)
```
Comment (vuvova) [inline extra/CMakeLists.txt:94]: Why?

File: man/CMakeLists.txt [hunk 2]
```diff
@@ -13,45 +13,55 @@
...
+                       mysql_install_db.1
+                       mysql_upgrade.1
+                       mysqld_multi.1 mysqld_safe.1
+                       mysqld_safe_helper.1)
+SET(MAN1_SERVER_NATIVE innochecksum.1 myisam_ftdump.1 myisamchk.1
+                       aria_chk.1 aria_dump_log.1 aria_ftdump.1 aria_pack.1 aria_read_log.1
+                       myisamlog.1 myisampack.1  mariadb-conv.1
+                       resolveip.1 mariadb-service-convert.1)
+SET(MAN1_SERVER ${MAN1_SERVER_COMPAT} ${MAN1_SERVER_NATIVE})
 SET(MAN8_SERVER mysqld.8)
```
Comment (vuvova) [inline man/CMakeLists.txt:31]: no compat/native for MAN8? what about mariadbd.8?

File: man/CMakeLists.txt [hunk 3]
```diff
@@ -13,45 +13,55 @@
...
+SET(MAN1_BACKUP mbstream.1 mariabackup.1)
+INSTALL(FILES ${MAN1_BACKUP} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT backup)
 
-INSTALL(FILES ${MAN1_CLIENT} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesClient)
+INSTALL(FILES ${MAN1_CLIENT_NATIVE} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesClient)
+INSTALL(FILES ${MAN1_CLIENT_COMPAT} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ClientSymlinks)
 INSTALL(FILES ${MAN1_DEVEL} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesDevelopment)
 INSTALL(FILES ${MAN1_TEST} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesTest)
 
+SET(MAN1_ROCKSDB mysql_ldb.1 myrocks_hotbackup.1)
```
Comment (vuvova) [inline man/CMakeLists.txt:61]: generally, all rocksdb related changes should be under storage/rocksdb

File: man/CMakeLists.txt [hunk 4]
```diff
@@ -13,45 +13,55 @@
...
+INSTALL(FILES ${MAN1_ROCKSDB} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT rocksdb-engine)
+INSTALL(FILES aria_s3_copy.1 DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT s3-engine)
```
Comment (vuvova) [inline man/CMakeLists.txt:63]: same for aria

**Review by vuvova — CHANGES_REQUESTED** [review]

File: man/CMakeLists.txt (hunk 3 above)
Comment (grooverdan) [reply man/CMakeLists.txt:61]: the macro that creates the mariadb-ldb symlink is there as was the original man page install.

File: extra/CMakeLists.txt (hunk 1 above)
Comment (grooverdan) [reply extra/CMakeLists.txt:94]: a) reduce Deb/RPM differences, Debian had this in client.
b) its man page was installed in client.

File: man/CMakeLists.txt (hunk 2 above)
Comment (grooverdan) [reply man/CMakeLists.txt:31]: its created by the `MARIADB_SYMLINK_MANPAGE` macro and placed in the `ManPagesServer` component.

File: man/CMakeLists.txt (hunk 4 above)
Comment (grooverdan) [reply man/CMakeLists.txt:63]: this is easy to move, but I thought there was some value in keeping all the man pages in one location.

File: man/CMakeLists.txt [hunk 5]
```diff
@@ -13,45 +13,55 @@
...
+SET(MAN1_TEST my_safe_process.1
+              mysql-stress-test.pl.1 mysql-test-run.pl.1 mysql_client_test.1
+              mysqltest.1 mysqltest_embedded.1 mysql_client_test_embedded.1)
 IF(NOT WITHOUT_SERVER)
-  INSTALL(FILES ${MAN1_SERVER} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesServer)
-  INSTALL(FILES ${MAN8_SERVER} DESTINATION ${INSTALL_MANDIR}/man8 COMPONENT ManPagesServer)
+  INSTALL(FILES ${MAN1_SERVER_NATIVE} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesServer)
+  IF (WITH_WSREP)
+    INSTALL(FILES ${MAN1_WSREP} DESTINATION ${INSTALL_MANDIR}/man1 COMPONENT ManPagesServer)
+  ENDIF()
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply man/CMakeLists.txt:49]: fix in update.

File: man/CMakeLists.txt (hunk 2 above)
Comment (vuvova) [reply man/CMakeLists.txt:31]: that won't do. `mariadbd.8` is just
```troff
.so man8/mysqld.8
```
while it should be the other way around

File: extra/CMakeLists.txt (hunk 1 above)
Comment (vuvova) [reply extra/CMakeLists.txt:94]: `resolve_stack_dump` is a server tool and it used to be in mariadb-server deb package too (up to 10.5, iirc).
But now many server tools are in the client package already, and both rpm and deb server packages require client anyway.
So ok, let's keep it in the client

**Comment by ottok** [comment]
I you merge #2396 to get Salsa-CI for up to 10.11, and then maybe @illuusio or @fauust can create the additional 11.0 upgrade scenario test on the 11.0 branch, which after this and other similar PRs can more easily be tested with proper CI support.

**Comment by illuusio** [comment]
> If you merge #2396 to get Salsa-CI up to 10.11, and then maybe @illuusio or @fauust can create the additional 11.0 upgrade scenario test on the 11.0 branch, which after this and other similar PRs can more easily be tested with proper CI support.

Do you mean to do it on Salsa?

**Comment by grooverdan** [comment]
Closing as @vuvova has picked and corrected the man page part of this in preview-11.0-preview.

@ottok @illuusio, noted about salsa 11.0 tests, might be prudent to wait until preview-11.0-preview is merged into 11.0 as its a big change.

**Pull request closed without merging**